scipy/stats/_continuous_distns.py

```diff
@@ -3431,18 +3431,13 @@ def g(n):
         g3 = g(3)
         g4 = g(4)
         g2mg12 = np.where(abs(c) < 1e-7, (c*np.pi)**2.0/6.0, g2-g1**2.0)
-        gam2k = _lazywhere(
-            abs(c) < 1e-7, (c,), 
-            f=lambda x: np.full_like(x, np.pi**2.0/6.0), 
-            f2=lambda x: sc.expm1(sc.gammaln(2.0*x+1.0)-2*sc.gammaln(x + 1.0))/x**2.0
-        )
+        def gam2k_f(c):
+            return sc.expm1(sc.gammaln(2.0*c+1.0)-2*sc.gammaln(c + 1.0))/c**2.0
+        gam2k = _lazywhere(abs(c) >= 1e-7, (c,), f=gam2k_f, fillvalue=np.pi**2.0/6.0)
```

Comment:
It's more natural to use `>=` here because that allows us to use `fillvalue`, which is simpler than `f=lambda x: np.full_like(x, np.pi**2.0/6.0)`. By defining a function `gam2k_f`, we separate the formula from the logic of `_lazywhere`.